src/functions-and-arrays.js

```diff
@@ -1,18 +1,84 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(x,y) {
+  if (x>=y) return x;
+  if (x<y) return y;
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord(wordArr) {
+  let l=0;
+  let longestWordIndex=0;
+  for (let i=0; i<wordArr.length; i++) {
+    if (wordArr[i].length>l) {
+      l=wordArr[i].length;
+      longestWordIndex=i;
+    }
+  }
+  return wordArr[longestWordIndex];
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers(numArr) {
+  let sum=0;
+for (num of numArr) {
+  sum += num;
+  //console.log(sum);
+}
+  return sum;
+}
+
+// Iteration #3.1
+function sum(arr) {
+  let replaceArr = [];
+  for (element of arr) {
+    console.log(typeof element);
+    if (typeof element == "number") {
+      replaceArr.push(element);
+    }
+    if (typeof element == "string") {
+      replaceArr.push(element.length);
+    }
+    if (element === false) {
+      replaceArr.push(0);
+    }
+    if (element === true) {
+      replaceArr.push(1);
+    }
+  }
```

Comment:
to solve this error
`should throw an error when unsupported data type (object or array) present in the array`
you can just check if the element type is == 'object' and throw the Error
```if (typeof element == 'object') throw new Error("Unsupported data type sir or ma'am");```